volmdlr/core.py

```diff
@@ -1476,33 +1478,46 @@ def babylon_data(self):
         return babylon_data
 
     @classmethod
-    def babylonjs_from_babylon_data(cls, babylon_data, page_name='Volmdlr model',
-                                    use_cdn=True, debug=False):
-
+    def babylonjs_from_babylon_data(cls, babylon_data, use_cdn=True,
+                                    debug=False):
         if use_cdn:
-            script = volmdlr.templates.babylon_unpacker_cdn_header#.substitute(name=page_name)
+            script = volmdlr.templates.babylon_unpacker_cdn_header  # .substitute(name=page_name)
         else:
-            script = volmdlr.templates.babylon_unpacker_embedded_header#.substitute(name=page_name)
+            script = volmdlr.templates.babylon_unpacker_embedded_header  # .substitute(name=page_name)
 
         script += volmdlr.templates.babylon_unpacker_body_template.substitute(
-                        babylon_data=babylon_data)
+            babylon_data=babylon_data)
+        return script
 
+    def babylonjs(self, page_name=None, use_cdn=True, debug=False):
+        babylon_data = self.babylon_data()
+        script = self.babylonjs_from_babylon_data(babylon_data, use_cdn=use_cdn,
+                                                  debug=debug)
         if page_name is None:
             with tempfile.NamedTemporaryFile(suffix=".html",
                                              delete=False) as file:
                 file.write(bytes(script, 'utf8'))
             page_name = file.name
-        else:
-            page_name += '.html'
-            with open(page_name, 'w')  as file:
-                file.write(script)
-
         webbrowser.open('file://' + os.path.realpath(page_name))
 
-    def babylonjs(self, page_name=None, use_cdn=True, debug=False):
+    def babylonjs_to_filepath(self, filepath: str, file_name: str = None,
+                              use_cdn=True, debug=False):
         babylon_data = self.babylon_data()
-        self.babylonjs_from_babylon_data(babylon_data, page_name=page_name,
-                                         use_cdn=use_cdn, debug=debug)
+        script = self.babylonjs_from_babylon_data(babylon_data, use_cdn=use_cdn,
+                                                  debug=debug)
+        if filepath.endswith('.html'):
+            with open(filepath, 'w') as file:
+                file.write(script)
+            return
+        if file_name is None:
```

Comment:
Better use tempfile to give us a tempfile name located in temp directory, by this it is deleted at computer restart

Reply:
do you mean doing this instead of creating a random name when no name is given?